### PR TITLE
[MISC] Improve optional types support

### PR DESCRIPTION
**Pull request by deusebio**

### CODE REVIEW

File: tests/unit/test_data_models.py [hunk 1]
```diff
@@ -201,3 +209,64 @@ def test_relation_databag_merged(self):
...
+    @parameterized.expand(
+        [
+            ("option-float", "1.0", float),
+            ("option-float", "1", float),
+            ("option-int", "1", int),
+            ("option-str", "1", str),
+            ("option-str", "test", str),
+        ]
+    )
+    def test_relation_databag_merged_with_option(self, option_key, option_value, _type):
```
Comment (welpaolo) [inline tests/unit/test_data_models.py:225]: Can you please add the docstring to new tests ?

**Review by welpaolo — APPROVED** [review]
Looks good to me! 
Just left a really minor comment!

File: tests/unit/test_data_models.py (hunk 1 above)
Comment (deusebio) [reply tests/unit/test_data_models.py:231]: Actually also old tests did not have doc strings :O I have updated them all in the data_models module

Comment (welpaolo) [reply tests/unit/test_data_models.py:231]: Nice! Thank you so much

File: tests/unit/test_data_models.py [hunk 2]
```diff
@@ -201,3 +215,68 @@ def test_relation_databag_merged(self):
...
+
+    @parameterized.expand(
+        [
+            ("option-float", "1.0", float),
+            ("option-float", "1", float),
+            ("option-int", "1", int),
+            ("option-str", "1", str),
+            ("option-str", "test", str),
+        ]
+    )
```
Comment (marcoppenheimer) [inline tests/unit/test_data_models.py:230]: **question:** Why not `pytest.mark.parameterize`?
If there wasn't a main reason, I would recommend we use that.

Comment (marcoppenheimer) [reply tests/unit/test_data_models.py:230]: Oh, it's `unittest`, right?

**Review by marcoppenheimer — APPROVED** [review]

File: tests/unit/test_data_models.py (hunk 2 above)
Comment (deusebio) [reply tests/unit/test_data_models.py:230]: yes, that's right. Actually, some tests used `unittests`, some used `pytest`. We should probably unify this at a certain point

**Pull request merged**